src/app/components/algorithms/implementation-view/implementation-nisq-analyzer-qpu-selection/implementation-nisq-analyzer-qpu-selection.component.ts

```diff
@@ -0,0 +1,278 @@
+            allowedProviders: [dialogResult.vendor],
+            circuitLanguage: this.nisqImpl.language,
+            circuitUrl: this.nisqImpl.fileLocation,
+            tokens: providerTokens,
+            refreshToken,
+            circuitName: this.nisqImpl.name,
+          };
+          this.nisqAnalyzerRootService
+            .selectQpuForCircuitFile1$Json({
+              simulatorsAllowed: dialogResult.simulatorAllowed,
+              circuitLanguage: this.nisqImpl.language,
+              circuitName: this.nisqImpl.name,
+              allowedProviders: [dialogResult.vendor],
+              tokens: providerTokens,
+              body: qpuSelectionDto,
+            })
+            .subscribe((job) => {
+              this.analyzerJob = job;
+              this.jobReady = job.ready;
+
+              this.utilService.callSnackBar(
+                'Successfully created analysis job "' + job.id + '".'
+              );
+              this.ngOnInit();
+
+              this.pollingAnalysisJobData = interval(2000)
+                .pipe(
+                  startWith(0),
+                  switchMap(() =>
+                    this.qpuSelectionService.getQpuSelectionJob({
+                      resId: this.analyzerJob.id,
+                    })
+                  )
+                )
+                .subscribe(
+                  (jobResult) => {
+                    this.analyzerJob = jobResult;
+                    this.jobReady = jobResult.ready;
+                    if (this.jobReady) {
+                      this.ngOnInit();
+                      this.analyzerResults = jobResult.qpuSelectionResultList;
+                      this.pollingAnalysisJobData.unsubscribe();
+                    }
+                  },
+                  () => {
+                    this.utilService.callSnackBar(
+                      'Error! Could not create analysis job.'
+                    );
+                  }
+                );
+            });
+        }
+      });
+  }
+
+  showAnalysisResult(analysisJob: QpuSelectionJobDto): boolean {
+    this.qpuSelectionService
+      .getQpuSelectionJob({ resId: analysisJob.id })
+      .subscribe((jobResult) => {
+        this.jobReady = jobResult.ready;
+        this.analyzerJob = jobResult;
+        this.analyzerResults = jobResult.qpuSelectionResultList;
+
+        for (const analysisResult of this.analyzerResults) {
+          this.showBackendQueueSize(analysisResult);
+          this.hasExecutionResult(analysisResult);
+        }
+      });
+    return true;
+  }
+
+  execute(analysisResult: QpuSelectionResultDto): void {
+    this.loadingResults[analysisResult.id] = true;
+    this.results = undefined;
+    this.executedAnalyseResult = analysisResult;
+    this.qpuSelectionService
+      .executeQpuSelectionResult({ resId: analysisResult.id })
+      .subscribe(
+        (results) => {
+          if (results.status === 'FAILED' || results.status === 'FINISHED') {
```

Comment:
Isn't the status realized in the form of a enum? I guess it should be an enum.

Reply:
see https://github.com/UST-QuAntiL/qc-atlas-ui/blob/582193894bfc9334b89a1fc7db4d539d6e76306f/generated/api-nisq/models/execution-result-dto.ts#L5
in the backend it is realized as enum.
But I cannot do results.status.FAILED...

Reply:
Okay, then we should create an issue that uses an enum here instead. Could you please formulate this as an issue?

Reply:
see https://github.com/UST-QuAntiL/qc-atlas-ui/issues/144